azure-core/azure/core/paging.py

```diff
@@ -0,0 +1,113 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import sys
+try:
+    from collections.abc import Iterator
+    xrange = range
+except ImportError:
+    from collections import Iterator
+
+from typing import Dict, Any, List, Callable, Optional, TYPE_CHECKING  # pylint: disable=unused-import
+
+from msrest.serialization import Deserializer
+
+if TYPE_CHECKING:
+    from .pipeline.transport import HttpResponse  # pylint: disable=unused-import
```

Comment:
Looks like this can be imported from `azure.pipeline` instead, I assume that is intended as preferable?

Reply:
hmm I think this is fine.
I usually do absolute imports, but Laurent tends to do relative ones - so whatever is your preference :)

Reply:
Oh, my question was vague. I was wondering specifically `azure.pipeline` vs more specific `azure.pipeline.transport`

Reply:
I went ahead and used the `.pipeline` in https://github.com/Azure/azure-sdk-for-python/pull/4873/commits/f442ee4a959c368be71286710105c121ae6177de as I believe the intent of that module is to collect public API from submodules